api/http/handlerfuncs.go

```diff
@@ -322,3 +324,32 @@ func subscriptionHandler(pub *events.Publisher[events.Update], rw http.ResponseW
 		}
 	}
 }
+
+// readWithTimeout reads from the reader until either EoF or the given maxDuration has been reached.
+func readWithTimeout(ctx context.Context, reader io.Reader, maxDuration time.Duration) ([]byte, error) {
```

Comment:
suggestion: Taking my above suggestion into account, this could be changed to a `readWithLimit` function where the limit is the maximum payload size we will allow.